runtime/sema/checker.go

```diff
@@ -2337,3 +2337,23 @@ func wrapWithOptionalIfNotNil(typ Type) Type {
 func (checker *Checker) CheckStatement(element ast.Statement) {
 	ast.AcceptStatement[struct{}](element, checker)
 }
+
+func (checker *Checker) checkStaticModifier(isStatic bool, position ast.HasPosition) {
+	if isStatic && !checker.Config.AllowStaticFunctions {
+		checker.report(
+			&InvalidStaticModifierError{
+				Range: ast.NewRangeFromPositioned(checker.memoryGauge, position),
+			},
+		)
+	}
+}
+
+func (checker *Checker) checkNativeModifier(isNative bool, position ast.HasPosition) {
+	if isNative && !checker.Config.AllowStaticFunctions {
```

Comment:
```suggestion
	if isNative && !checker.Config.AllowNativeFunctions {
```

Reply:
ah just noticed, maybe rename this to `AllowNativeDeclarations` since it checks both fields and functions. Same for the other one too.

Reply:
Good catch! Fixed use and improved naming in a409088c56b1adc81f4964047521b3ca6937ee7f